modules/fedimint-ln-client/src/lib.rs

```diff
@@ -1357,6 +1358,79 @@ impl LightningClientModule {
         }))
     }
 
+    /// Scan unspent incoming contracts for a payment hash that matches a
+    /// tweaked keys in the `indices` vector
+    pub async fn scan_receive_for_user_tweaked(
+        &self,
+        key_pair: KeyPair,
+        indices: Vec<u64>,
+    ) -> anyhow::Result<Amount> {
+        for i in indices {
+            let key_pair_tweaked = tweak_user_secret_key(&self.secp, key_pair, i);
+            if let Ok(amount) = self.scan_receive_for_user(key_pair_tweaked).await {
+                return Ok(amount);
```

Comment:
Is returning early really what we want to do here? Won't we be missing contracts or is this function meant to be called multiple times with the same list of indices?

Reply:
Ah yeah, makes more sense to keep a total and return at the end

Reply:
I'll change this to return `Vec<OperationId>`